tests/setup_test.go

```diff
@@ -1030,26 +1031,113 @@ func TestUpdateProject(t *testing.T) {
 	t.Parallel()
 
 	tests := []struct {
-		name               string
-		passVarsToForkedPR bool
-		expectedPre        bool
-		expectedPost       bool
+		name                        string
+		passVarsToForkedPR          bool
+		expectedPre                 bool
+		expectedPost                bool
+		membersCanPerformRunActions bool
 	}{
 		{
-			name:               "test project update with pass-vars-to-forked-pr true",
-			passVarsToForkedPR: true,
-			expectedPre:        false,
-			expectedPost:       true,
+			name:                        "test project update with pass-vars-to-forked-pr true and membersCanPerformRunActions false ",
+			passVarsToForkedPR:          true,
+			expectedPre:                 false,
+			expectedPost:                true,
+			membersCanPerformRunActions: false,
 		},
 		{
-			name:               "test project update with pass-vars-to-forked-pr false",
-			passVarsToForkedPR: false,
-			expectedPre:        false,
-			expectedPost:       false,
+			name:                        "test project update with pass-vars-to-forked-pr false and membersCanPerformRunActions false",
+			passVarsToForkedPR:          false,
+			expectedPre:                 false,
+			expectedPost:                false,
+			membersCanPerformRunActions: false,
 		},
 	}
 
 	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			dir := t.TempDir()
+
+			ctx, cancel := context.WithCancel(context.Background())
+			defer cancel()
+
+			sc := setup(ctx, t, dir, withGitea(true))
+			defer sc.stop()
+
+			giteaAPIURL := fmt.Sprintf("http://%s:%s", sc.gitea.HTTPListenAddress, sc.gitea.HTTPPort)
+
+			giteaToken, token := createLinkedAccount(ctx, t, sc.gitea, sc.config)
+
+			giteaClient, err := gitea.NewClient(giteaAPIURL, gitea.SetToken(giteaToken))
+			if err != nil {
+				t.Fatalf("unexpected err: %v", err)
+			}
+
+			gwClient := gwclient.NewClient(sc.config.Gateway.APIExposedURL, token)
+
+			_, project := createProject(ctx, t, giteaClient, gwClient)
+			if project.PassVarsToForkedPR != tt.expectedPre {
+				t.Fatalf("expected PassVarsToForkedPR %v, got %v (pre-update)", tt.expectedPre, project.PassVarsToForkedPR)
+			}
+			if project.MembersCanPerformRunActions != tt.membersCanPerformRunActions {
+				t.Fatalf("expected MembersCanPerformRunActions %v, got %v (pre-update)", tt.membersCanPerformRunActions, project.MembersCanPerformRunActions)
+			}
+			project = updateProject(ctx, t, giteaClient, gwClient, project.ID, tt.passVarsToForkedPR)
+			if project.PassVarsToForkedPR != tt.expectedPost {
+				t.Fatalf("expected PassVarsToForkedPR %v, got %v (port-update)", tt.expectedPost, project.PassVarsToForkedPR)
+			}
+			if project.MembersCanPerformRunActions != tt.membersCanPerformRunActions {
+				t.Fatalf("expected MembersCanPerformRunActions %v, got %v (port-update)", tt.membersCanPerformRunActions, project.MembersCanPerformRunActions)
+			}
+		})
+	}
```

Comment:
This mix of tests matrix and single sub tests is confusing. Also the pre/post stuff could be removed. Probably the 2 tests in the matrix could be just changed to standard sub tests.